hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java

```diff
@@ -139,10 +146,126 @@ void nonLazyCreateCandidateMustHaveReceiver() {
         assertAbortsWith(wellKnownRelayedHapiCall(0), INVALID_CONTRACT_ID);
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    void ethCreateHappyPathAsExpected() {
+        final var inOrder = inOrder(worldUpdater, frameBuilder, frameRunner, gasCharging, senderAccount);
+
+        givenSenderAccount();
+        givenRelayerAccount();
+
+        final var context = wellKnownContextWith(code, blocks);
+        final var transaction = wellKnownRelayedHapiCreate();
+
+        given(gasCharging.chargeForGas(senderAccount, relayerAccount, context, worldUpdater, transaction))
+                .willReturn(CHARGING_RESULT);
+        given(senderAccount.getAddress()).willReturn(EIP_1014_ADDRESS);
+        given(senderAccount.getNonce()).willReturn(NONCE);
+        final var expectedToAddress = Address.contractAddress(EIP_1014_ADDRESS, NONCE);
+        given(frameBuilder.buildInitialFrameWith(
+                        transaction,
+                        worldUpdater,
+                        context,
+                        config,
+                        EIP_1014_ADDRESS,
+                        expectedToAddress,
+                        CHARGING_RESULT.intrinsicGas()))
+                .willReturn(initialFrame);
+        given(frameRunner.runToCompletion(
+                        transaction.gasLimit(), initialFrame, tracer, messageCallProcessor, contractCreationProcessor))
+                .willReturn(SUCCESS_RESULT);
+        given(initialFrame.getSelfDestructs()).willReturn(Set.of(NON_SYSTEM_LONG_ZERO_ADDRESS));
+
+        final var result =
+                subject.processTransaction(transaction, worldUpdater, () -> feesOnlyUpdater, context, tracer, config);
+
+        inOrder.verify(worldUpdater).setupAliasedCreate(Address.ZERO, expectedToAddress);
+        inOrder.verify(senderAccount).incrementNonce();
+        inOrder.verify(gasCharging).chargeForGas(senderAccount, relayerAccount, context, worldUpdater, transaction);
+        inOrder.verify(frameBuilder)
+                .buildInitialFrameWith(
+                        transaction,
+                        worldUpdater,
+                        context,
+                        config,
+                        EIP_1014_ADDRESS,
+                        expectedToAddress,
+                        CHARGING_RESULT.intrinsicGas());
+        inOrder.verify(frameRunner)
+                .runToCompletion(
+                        transaction.gasLimit(), initialFrame, tracer, messageCallProcessor, contractCreationProcessor);
+        inOrder.verify(gasCharging)
+                .maybeRefundGiven(
+                        GAS_LIMIT - SUCCESS_RESULT.gasUsed(),
+                        CHARGING_RESULT.relayerAllowanceUsed(),
+                        senderAccount,
+                        relayerAccount,
+                        context,
+                        worldUpdater);
+        inOrder.verify(worldUpdater).deleteAccount(NON_SYSTEM_LONG_ZERO_ADDRESS);
```

Comment:
Is this really part of a "happy path" test?  Or is it - maybe - here because it's convenient to test it here without having to spin up another largely identical test?  Or?

Reply:
Also same two questions for `hapiCreateHappyPathAsExpected` below.